### PR TITLE
🏗🐛 Remove JSDoc from third party code before compiling it

### LINKED ISSUE #21828: `third_party/` code must not be treated as closure code
AMP components like `extensions/amp-subscriptions/` include third party code that is checked in to our repo under `third_party/`. When we run the type checker via `gulp check-types` or minify code via `gulp dist`, closure compiler treats the code in `third_party/` as if it were closure code.

- This has always been the case, but we've never realized it
- This is risky because code in `third_party/` may interfere with compilation optimizations around inlining, devirtualization, minification, etc.

@dvoytenko recently discovered this due to following events:

1. The version of closure compiler used by AMP was upgraded from an old version to `v20190301` (https://github.com/ampproject/amphtml/pull/21618)
2. The latest compiler version has a bug where it does not properly detect relative path annotations (https://github.com/google/closure-compiler/issues/3041)
3. The latest update to `third_party/subscriptions-project/swg.js` [contains](https://github.com/ampproject/amphtml/pull/21825/commits/b3cf916f7dfa25f2281ecfa22b7b62eb1d3a7ee8#diff-94eeef538054ed1c247850e89073723bR12779) such an annotation
4. As a result, the type checker [complains](https://travis-ci.org/ampproject/amphtml/jobs/518900497#L385) about a bad closure type annotation in third party code.

To fix this, we must no longer treat code in `third_party/` as closure code.
- One way to do this is to strip away all JSDoc comments before passing the code to closure compiler.
- There may be a way to pass in options to closure compiler to treat certain files as non-closure code.

This issue is meant to discuss and track a fix.

### PR DESCRIPTION
**Pull request by rsimha**
This PR attempts to fix #21828 by removing JSDoc comments from  code in `third_party/`.

The code looks elegant, but I don't think it's going to work as written, because `gulp-closure-compiler` ignores the contents of the file stream passed into it, and simply uses the name of the file in order to read it from disc.

:frowning_face:

### CODE REVIEW

File: build-system/tasks/compile.js [hunk 1]
```diff
@@ -421,6 +423,7 @@ function compile(entryModuleFilenames, outputDir, outputFilename, options) {
     }
 
     let stream = gulp.src(srcs)
+        .pipe(gulpIf(isThirdPartyFile, gulpStringReplace('/**', '/* ')))
```
Comment (jridgewell) [inline build-system/tasks/compile.js]: We already have a gulp replace, see https://github.com/ampproject/amphtml/pull/21830/files#diff-b2216d09c55c9ee1f2f2c879bee3b619R439

Also, can you verify that this is working correctly? I tried to do replacement before running through closure in https://github.com/ampproject/amphtml/pull/21818, and it didn't work.

Comment (rsimha) [reply build-system/tasks/compile.js]: @jridgewell See description.

**Comment by rsimha** [comment]
I'll reattempt this fix after #21868 is merged.

**Comment by rsimha** [comment]
This PR can be reattempted after #21868 is merged.

**Pull request closed without merging**